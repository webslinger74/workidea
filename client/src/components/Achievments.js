import React from 'react';


const Achievements = () => {
    return (
        <div>
            Achievements are cool
        </div>
      );
}
 
export default Achievements;