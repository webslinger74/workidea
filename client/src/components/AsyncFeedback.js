import Loadable from 'react-loadable';
import React from 'react';

const LoadingComponent = () => <h3>Please Wait...</h3>

const FeedbackComponentPromise = () => {
    return import('./Feedback');
}

const AsyncFeedback = Loadable({
    loader:FeedbackComponentPromise,
    loading:LoadingComponent
});

export default AsyncFeedback;
