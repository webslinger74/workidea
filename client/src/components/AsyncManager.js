import Loadable from 'react-loadable';
import React from 'react';

const LoadingComponent = () => <h3>Please Wait...</h3>

const ManagerComponentPromise = () => {
    return import('./Manager');
}

const AsyncManager = Loadable({
    loader:ManagerComponentPromise,
    loading:LoadingComponent
});

export default AsyncManager;