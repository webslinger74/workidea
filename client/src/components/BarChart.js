import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as d3 from 'd3';
import d3Tip from 'd3-tip';
 

class BarChart extends Component {
    constructor(props) {
        super(props);
        this.state = {
        }
    }


    componentDidMount(){
        this.drawChart();
    }

    componentDidUpdate(prevProps){
        if(this.props.data !== prevProps.data){
            this.drawChart();
        }

    }

    drawChart = () => {

       
        const data = this.props.data;

        const max = d3.max(data, d => d.amount);

        const margin = {
            top:20, right:20, bottom:100, left:100
        }

        const graphWidth = 600 - margin.right - margin.left;
        const graphHeight = 500 - margin.top - margin.bottom;


        const y = d3.scaleLinear()
            .domain([0,max])
            .range([graphHeight,0]);

        const x = d3.scaleBand()
            .domain(data.map(item => item.name))
            .range([0,500])
            .paddingInner(0.2)
            .paddingOuter(0.2);

       

      //  const svg = d3.select ('#barChart').append("svg").attr("width", 700).attr("height", 600);

        let u = d3.select(this.svgElememt)
        
        let gr = d3.select(this.grElement)
        
            gr.attr('width', graphWidth)
            .attr('height', graphHeight)
            .attr('transform', `translate(${margin.left}, ${margin.top})`)


        let grX = d3.select(this.grXElement);
        let grY = d3.select(this.grYElement);

        const xAxisGroup = grX
                .attr('transform', `translate(0, ${graphHeight})`);
        const yAxisGroup = grY;
                
        

        const xAxis = d3.axisBottom(x);
        const yAxis = d3.axisLeft(y);

        xAxisGroup.call(xAxis);
        yAxisGroup.call(yAxis);

        //up to here is controlled by React then kicks in the D3 parts

         
            //rects already in DOM
        const rects = gr.selectAll('rect')
            .data(data);


          /* Initialize tooltip */
       const tip = d3Tip().attr('class', 'd3-tip').html(function(d) { return d.amount; });
       /* Invoke the tip in the context of your visualization */
             rects.call(tip)   
      

            rects.exit().remove();
        
           rects.attr("fill", this.props.color)
            .attr('class', "bars")
            .attr("x", d => x(d.name))
            .attr("y", graphHeight)
            .attr("height", 0)
            .attr("width", x.bandwidth)
            .on('mouseover', tip.show)
            .on('mouseout', tip.hide)     
            .transition().duration(4000)
            .attr("height", d => graphHeight - y(d.amount))
            .attr("y", d => y(d.amount));
   /* Invoke the tip in the context of your visualization */
   rects.call(tip)   
      
   
//enter selection////////////////
        rects.enter()
        .append('rect')
                .attr("fill", this.props.color)
                .attr('class', "bars")
                .attr("x", d => x(d.name))
                .attr("y", graphHeight)
                .attr("height", 0)
                .attr("width", x.bandwidth)
                .on('mouseover', tip.show)
                .on('mouseout', tip.hide)     
                .transition().duration(4000)
                  .attr("height", d => graphHeight - y(d.amount))
                  .attr("y", d => y(d.amount));

   /* Invoke the tip in the context of your visualization */
   rects.call(tip)   
      
          
    }

      render(){
          
    return (

        

        <div>
                   
            <div className="dynamicCharts">
            <svg width={this.props.width} height={this.props.height} ref={element => this.svgElememt = element}>
                <g ref={element => this.grElement = element}>
                <g ref={element => this.grXElement = element}></g>
                <g ref={element => this.grYElement = element}></g>
                
                </g>
            </svg>
          
            </div>
         
            

       </div>
    
      
      )
    }
}



const actions = {
}

const mapStateToProps = (state) => ({

})
 
export default connect(mapStateToProps, actions)(BarChart);