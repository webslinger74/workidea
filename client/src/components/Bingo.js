import React, { Component } from 'react';
import { getBingoNumbers } from '../actions/sportsActions';
import { connect } from 'react-redux';
import Sports from './Sports';

class Bingo extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

         componentDidMount(){
            this.props.getBingoNumbers();
         }

      render(){
        const { bingo } = this.props;
        console.log(bingo, "bingo");

    return (
            <Sports>
        <div>    
        <div className="sportsItem">
        <h1>Pay Day Bingo</h1>
            <p>Come and Play Bingo Every Pay Day on Site. Â£1 a ticket with Cash prizes to be Won every Month</p>  
            <p>Local reps will be selling tickets near you in the lead up to Pay day</p>
            <p>Rules - Email BST as soon as you have all the numbers on  your purchased ticket</p>
            <p>Email with the number that you completed your card with !</p>
            <p>GOOD LUCK</p>  
            <div id="bingocard">               
                   <div className="cont">
                   { bingo && bingo.map(element => {
                    
                    return (<div id={`bingoCardInner${element.number}`} key={element._id}> {element.number}
                    </div> )
                   })
                   }
                    
                    
        </div>
        
        </div>
        </div>
        <div className="dividers2"></div>
        <div className="sportsItem">
    
        </div>
      
      
        </div>
        </Sports>
      )
    }
}

const actions = {
    getBingoNumbers
}

const mapStateToProps = (state) => ({
    bingo:state.sports.bingoNumbers
})
 
export default connect(mapStateToProps, actions)(Bingo);