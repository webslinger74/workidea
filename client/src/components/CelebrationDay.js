import React, { Component } from 'react';
import { connect } from 'react-redux';
import Sports from './Sports';
import ReactHtmlParser from 'react-html-parser';
import { getCelebrationDay, deleteCelebrationDay } from '../actions/sportsActions';

class CelebrationDay extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

    convertStringMessageToHtml = (string) => {
        const html = string;
        return <div>{ReactHtmlParser(html)}</div>
}

showMessages = (messages) => {

    
   if(messages && messages[0]) {
   return (
         <div className="latestSportsMessage" key={messages[0]._id}>
       <div>
            <h3 className="latestMessageTitle"> {messages[0].title} </h3> </div>
            
            {messages[0].images && messages[0].images.length > 0 ?
                       (<div className="imgHolder">
                            
                               <img className="messageBoardImgFront" src={messages[0].images[0].url}></img>
                       </div>) : null
                }
              
              
            
                <div>
       {messages[0].message ? 
            this.convertStringMessageToHtml(messages[0].message):
            null}                    
            </div>
            <div>
            <h3 className="authorStamp"> {messages[0].author} - {messages[0].createdAt}  </h3>
        </div>
        {this.props.auth ?
        <div style={{paddingRight:"25px", width:"150px", fontSize:"20px", backgroundColor:"black", color:"white"}} onClick={()=> this.props.deleteCelebrationDay({id:messages[0]._id}, messages[0].images)} className="delete">Delete Celebration Day</div>
        : null } </div>
          
     )
  
       } else {
           return (
               <div>No Celebration Day is scheduled currently, Try again later</div>
           )
       }
    }

       componentDidMount(){
          this.props.getCelebrationDay();
        }

       

      render(){
        const { celebrationDay } = this.props;
    return (
            <Sports>
        
        <div className="sportsItem2"> 
        <div id="celebrationBackground">   
            <div  style={{backgroundColor:'gold'}} className="centredLatest">Celebration Day</div>
            
                {this.showMessages(celebrationDay)}
                 
            </div>
        <div className="dividers2"></div>
        <div className="sportsItem">
    
        </div>
      
      
        </div>
        </Sports>
      )
    }
}

const actions = {
    getCelebrationDay,
    deleteCelebrationDay
}

const mapStateToProps = (state) => ({
    celebrationDay:state.sports.celebrationDay,
    auth:state.auth.isAuthenticated
})
 
export default connect(mapStateToProps, actions)(CelebrationDay);