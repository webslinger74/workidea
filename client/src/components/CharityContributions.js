import React, { Component } from 'react';
import { connect } from 'react-redux';
import Sports from './Sports';
import { getCharity, deleteCharity } from '../actions/sportsActions';

class CharityContributions extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }

         componentDidMount(){
             this.props.getCharity();
         }

         getGrandTotal = () => {
             var grandTotal = 0;
             this.props.charities.map( (charity) => {
                 grandTotal += parseInt(charity.amount)
             })
             return grandTotal;
         }

      render(){
            const { charities } = this.props;
    return (
            <Sports>
        <div className="charityBackground">    
        <div>
        <h1  style={{backgroundColor:'gold'}}>Money Raised this Year For Charity: £{(this.props.charities && this.props.charities) ? this.getGrandTotal() : 0.00}</h1>
           
            <p>Please Check out how much YOU have helped the S&Social raise through site events this year!</p>  

                        {charities && charities ? (
                            <div className="charityContainer">
                                {charities.map((charity) => {
                                  return  (
                                        <div key={charity._id} className="charityItems">
                                         Total Raised for: <span style={{fontSize:'40px', color:'green'}}>{charity.charity}</span> = <strong style={{fontSize:'40px'}}>£{charity.amount}</strong> 
                                         <div onClick={()=> this.props.deleteCharity({id:charity._id})}><span style={{fontSize:'12px'}}>DELETE</span></div>
                                        </div>
                                    )
                                }

                                )}

                            </div>
                        ):null}





        </div>
        <div className="dividers2"></div>
        <div className="sportsItem">
    
        </div>
      
      
        </div>
        </Sports>
      )
    }
}

const actions = {
    getCharity,
    deleteCharity
}

const mapStateToProps = (state) => ({
    charities:state.sports.charities
})
 
export default connect(mapStateToProps, actions)(CharityContributions);