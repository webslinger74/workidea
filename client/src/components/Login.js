import React, {Component} from 'react';
import axios from 'axios';
import { connect } from 'react-redux';
import { loginUser } from '../actions/userActions';
import TextFieldGroup from '../Inputs/TextFieldGroup';

class Login extends Component {
constructor(){
    super();
    this.state = {
        email:"",
        password:"",
        errors: {}
    };
}

  componentDidMount(){
    if(this.props.auth.isAuthenticated){
      this.props.history.push('/dashboard')
    }
  }

  componentWillReceiveProps(nextProps) {
    if(nextProps.auth.isAuthenticated) {
      this.props.history.push('/user/dashboard');
    }

    if(nextProps.errors){
      this.setState({
        ...this.state,
        errors:nextProps.errors
      })
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  onSubmit = (e) => {
    e.preventDefault();

    const UserData = {
      email: this.state.email,
      password: this.state.password
    };
  
  this.props.loginUser(UserData);
   
  }

    render() { 
        const {errors} = this.state;

        return (
          <div className="page_wrapper">
          <div className="container">
                  <div className="signin_wrapper">

                <form onSubmit={this.onSubmit}>
                <h1>LOGIN WITH YOUR DETAILS</h1>
                <TextFieldGroup 
                    type="email"
                    placeholder="Email Address"
                    name="email"
                    error={errors.email}
                    value={this.state.email}
                    onChange={this.onChange}
                  />
                     <TextFieldGroup 
                    type="password"
                    placeholder="Password"
                    name="password"
                    error={errors.password}
                    value={this.state.password}
                    onChange={this.onChange}
                  />             

                    { this.state.formError ?
                        <div className="error_label">
                            Please check your data
                        </div>
                    :null}
                  
                  <input type="submit" className="btn btn-info btn-block mt-4" />
                </form>
            </div>
</div>
</div>
          )
    }
}

const actions = {
  loginUser
}

const mapStateToProps = (state) => {
  return {
      auth: state.auth,
      errors:state.errors
}}


 
export default connect(mapStateToProps,actions)(Login);