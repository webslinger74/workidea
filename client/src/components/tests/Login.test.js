import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../components/Login';
import setUpTest from '../../setUpTest';

it('should render the Login component', () => {
    const loginUser = jest.fn();
    const auth = {isAuthenticated:false};

         expect(shallow(<Login loginUser={loginUser} auth={auth}/>)).toMatchSnapshot();
            expect(loginUser).toHaveBeenCalledTimes(0);  // only  called on submit so should not be called
          })
