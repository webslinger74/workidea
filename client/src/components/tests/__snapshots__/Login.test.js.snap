// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Login component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    auth={
      Object {
        "isAuthenticated": false,
      }
    }
    loginUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <div
          className="signin_wrapper"
        >
          <form
            onSubmit={[Function]}
          >
            <h1>
              ADMIN USER -- LOGIN
            </h1>
            <TextFieldGroup
              name="staffnumber"
              onChange={[Function]}
              placeholder="Staff Number"
              type="text"
              value=""
            />
            <TextFieldGroup
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <input
              type="submit"
              value="Login"
            />
          </form>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="signin_wrapper"
        >
          <form
            onSubmit={[Function]}
          >
            <h1>
              ADMIN USER -- LOGIN
            </h1>
            <TextFieldGroup
              name="staffnumber"
              onChange={[Function]}
              placeholder="Staff Number"
              type="text"
              value=""
            />
            <TextFieldGroup
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <input
              type="submit"
              value="Login"
            />
          </form>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
          >
            <h1>
              ADMIN USER -- LOGIN
            </h1>
            <TextFieldGroup
              name="staffnumber"
              onChange={[Function]}
              placeholder="Staff Number"
              type="text"
              value=""
            />
            <TextFieldGroup
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <input
              type="submit"
              value="Login"
            />
          </form>,
          "className": "signin_wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                ADMIN USER -- LOGIN
              </h1>,
              <TextFieldGroup
                name="staffnumber"
                onChange={[Function]}
                placeholder="Staff Number"
                type="text"
                value=""
              />,
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
              null,
              <input
                type="submit"
                value="Login"
              />,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ADMIN USER -- LOGIN",
              },
              "ref": null,
              "rendered": "ADMIN USER -- LOGIN",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "staffnumber",
                "onChange": [Function],
                "placeholder": "Staff Number",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Login",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <div
            className="signin_wrapper"
          >
            <form
              onSubmit={[Function]}
            >
              <h1>
                ADMIN USER -- LOGIN
              </h1>
              <TextFieldGroup
                name="staffnumber"
                onChange={[Function]}
                placeholder="Staff Number"
                type="text"
                value=""
              />
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <input
                type="submit"
                value="Login"
              />
            </form>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="signin_wrapper"
          >
            <form
              onSubmit={[Function]}
            >
              <h1>
                ADMIN USER -- LOGIN
              </h1>
              <TextFieldGroup
                name="staffnumber"
                onChange={[Function]}
                placeholder="Staff Number"
                type="text"
                value=""
              />
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <input
                type="submit"
                value="Login"
              />
            </form>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              onSubmit={[Function]}
            >
              <h1>
                ADMIN USER -- LOGIN
              </h1>
              <TextFieldGroup
                name="staffnumber"
                onChange={[Function]}
                placeholder="Staff Number"
                type="text"
                value=""
              />
              <TextFieldGroup
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <input
                type="submit"
                value="Login"
              />
            </form>,
            "className": "signin_wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  ADMIN USER -- LOGIN
                </h1>,
                <TextFieldGroup
                  name="staffnumber"
                  onChange={[Function]}
                  placeholder="Staff Number"
                  type="text"
                  value=""
                />,
                <TextFieldGroup
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />,
                null,
                <input
                  type="submit"
                  value="Login"
                />,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ADMIN USER -- LOGIN",
                },
                "ref": null,
                "rendered": "ADMIN USER -- LOGIN",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "staffnumber",
                  "onChange": [Function],
                  "placeholder": "Staff Number",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "error": undefined,
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "submit",
                  "value": "Login",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
